### PR TITLE
Remove licensing subproject, move doc to vendor.md

### PR DESCRIPTION
**Pull request by nikhita**
Ref: https://github.com/kubernetes/sig-release/pull/1412#discussion_r556191521

/cc @dims @justaugustus @saschagrunert @hasheddan @alejandrox1

### CODE REVIEW

File: contributors/devel/sig-architecture/vendor.md [hunk 1]
```diff
@@ -164,10 +164,12 @@ Additionally:
 - If this is all good, approve, but don't LGTM, unless you also do code review
   or unless it is trivial (e.g. moving from k/k/pkg/utils -> k/utils).
 
+Licenses for dependencies are specified by the CNCF [Allowlist Policy].
```
Comment (saschagrunert) [inline contributors/devel/sig-architecture/vendor.md:167]: Should `[Allowlist Policy]` be a link to some external reference?

Comment (nikhita) [reply contributors/devel/sig-architecture/vendor.md:167]: Fixed :+1:

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
/approve
/lgtm

feel free to remove hold @nikhita

**Pull request merged**